pkg/splunk/enterprise/util.go

```diff
@@ -32,6 +34,55 @@ import (
 // kubernetes logger used by splunk.enterprise package
 var log = logf.Log.WithName("splunk.enterprise")
 
+// regex to extract the region from the s3 endpoint
+var regionRegex = ".*.s3[-,.](?P<region>.*).amazonaws.com"
+
+//getRegion extracts the region from the endpoint field
+func getRegion(endpoint string) string {
+	appS3Endpoint := endpoint
+	pattern := regexp.MustCompile(regionRegex)
+	return pattern.FindStringSubmatch(appS3Endpoint)[1]
+}
+
+// GetRemoteStorageClient returns the corresponding S3Client
+func GetRemoteStorageClient(client splcommon.ControllerClient, cr splcommon.MetaObject, appFrameworkRef *enterprisev1.AppFrameworkSpec, vol *enterprisev1.VolumeSpec, location string) (splclient.S3Client, error) {
+
+	scopedLog := log.WithName("GetRemoteStorageClient").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	//use the provider name to get the corresponding function pointer
+	getClient, _ := splclient.S3Clients[vol.Provider]
```

Comment:
Do we need to error out here if the entry for `vol.Provider` does not exist?

Reply:
I believe we are already checking that as part of the validation of cr and if this field is empty we won't even come this far.